src/openrct2/object/TerrainSurfaceObject.h

```diff
@@ -53,6 +53,7 @@ class TerrainSurfaceObject final : public Object
     uint8_t Rotations{};
     money32 Price{};
     TERRAIN_SURFACE_FLAGS Flags{};
+    uint8_t MapColours[2];
```

Comment:
I think we need to create a new typedef for palette index, otherwise it is difficult to know whether we are talking about the 32 possible user colours or the palette index.